grape/fault_diagnosis.py

```diff
@@ -120,6 +119,143 @@ def check_before(self):
             'indegree_centrality', 'outdegree_centrality', 'degree_centrality']
         self.update_output(centrality_fields)
 
+    def fitness_evaluation(self, individual, perturbed_nodes,
+        initial_condition):
+        """
+
+        Evaluation of fitness on individual.
+        The individual is a list of conditions for the graph switches
+        (True or False).
+        Edges towards successors are removed if the switch state
+        is set to 'False'.
+
+        :param list perturbed_nodes: nodes(s) involved in the
+            perturbing event.
+        :param dict initial_condition: initial status (boolean) for the graph
+            switches.
+        """
+
+        n_actions = np.sum(np.not_equal(list(initial_condition.values()),
+            individual))
+
+        T = GeneralGraph()
+        T.add_nodes_from(self.G) 
+        for (u, v, d) in self.G.edges(data=True):
+            T.add_edge(u, v, father_condition=d['father_condition'],
+                weight=d['weight'])
+        nx.set_node_attributes(T, self.G.initial_service,
+            name='initial_service')
+        nx.set_node_attributes(T, self.G.perturbation_resistant,
+            name='perturbation_resistant')
+        nx.set_node_attributes(T, self.G.type, name='type')
```

Comment:
Is this just a copy of the original graph? In the case, `T = self.copy()` should do the job

Reply:
It is a copy of the original graph, in the sense same number of nodes and same edges, but with very few attributes, namely `initial_service`, `perturbation_resistant` and `type`.